### PR TITLE
Introduce `StdOutToMetadataFeedback` and `StdErrToMetadataFeddback`

### PR DESCRIPTION
**Pull request by l4yton**

### CODE REVIEW

File: libafl/src/feedbacks/stdio.rs [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+impl Named for StdOutFeedback {
+    #[inline]
+    fn name(&self) -> &str {
+        self.name.as_str()
+    }
+}
+
+impl StdOutFeedback {
+    /// Creates a new [`StdOutFeedback`].
```
Comment (domenukk) [inline libafl/src/feedbacks/stdio.rs:98]: I would either document that the name is used to look up the observer, or leave out this constructor completely

File: libafl/src/feedbacks/stdio.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+pub struct StdErrFeedback {
+    name: String,
+}
+
+impl<S> Feedback<S> for StdErrFeedback
+where
+    S: State,
+{
+    #[allow(clippy::wrong_self_convention)]
+    fn is_interesting<EM, OT>(
```
Comment (domenukk) [inline libafl/src/feedbacks/stdio.rs:136]: `#[inline]`

File: libafl/src/feedbacks/stdio.rs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+impl Named for StdErrFeedback {
+    #[inline]
+    fn name(&self) -> &str {
+        self.name.as_str()
+    }
+}
+
+impl StdErrFeedback {
+    /// Creates a new [`StdErrFeedback`].
```
Comment (domenukk) [inline libafl/src/feedbacks/stdio.rs:195]: Same as above: add documentation or leave out completely

**Review by domenukk — APPROVED** [review]

**Comment by domenukk** [comment]
I wonder if this should be called feedback, that name sounds more like it will look for new stdout or stderr as ... feedback.
Or maybe a name like `StdOutToMetadataFeedback` or similar?

**Comment by l4yton** [comment]
`StdOutToMetadataFeedback` sounds good.
Since storing the stdout/stderr for every testcase may be a bit expensive, I thought about adding an option like `store_on_crash_only`, what do you think?

**Comment by l4yton** [comment]
Or is that even possible 🤔

**Comment by domenukk** [comment]
You can just only use it as Objective (not normal feedback), and then `and`it with the other options, like
`feedback_and_fast!(WhateverFeedback, std_out_to_metadata_feedback)` then it'll only be called if it's an objective

**Comment by l4yton** [comment]
Makes sense, thank you!

File: libafl/src/feedbacks/mod.rs [hunk 4]
```diff
@@ -23,6 +23,8 @@ pub use new_hash_feedback::NewHashFeedbackMetadata;
 
 #[cfg(feature = "nautilus")]
 pub mod nautilus;
+#[cfg(feature = "std")]
+pub mod stdio;
```
Comment (l4yton) [inline libafl/src/feedbacks/mod.rs:27]: Should I add `pub use stdio::{StdOutToMetadataFeedback, StdErrToMetadataFeedback};` here?

Comment (domenukk) [reply libafl/src/feedbacks/mod.rs:27]: Don't think it'll be used _that_ often, right? But up to you

File: libafl/src/feedbacks/stdio.rs [hunk 5]
```diff
@@ -15,7 +15,7 @@ use crate::{
     Error, HasMetadata,
 };
 
-/// Metadata for `StdOutFeddback`,
+/// Metadata for `StdOutToMetadataFeedback`.
```
Comment (domenukk) [inline libafl/src/feedbacks/stdio.rs:18]:
```rust
Metadata for [`StdOutToMetadataFeedback`].
```
would get us the link in the docs

(Does the metadata have to be `pub` at all?)

Comment (domenukk) [reply libafl/src/feedbacks/stdio.rs:18]: Ah nevermind it needs to be pub if you want to read from it? Unless you got a function to access it

File: libafl/src/feedbacks/mod.rs (hunk 4 above)
Comment (l4yton) [reply libafl/src/feedbacks/mod.rs:27]: Yeah, will leave it as is then.

File: libafl/src/feedbacks/stdio.rs (hunk 5 above)
Comment (l4yton) [reply libafl/src/feedbacks/stdio.rs:18]: Yes, I could `impl` a getter function for the field, but not a huge fan of that tbh

**Pull request merged**